salt/modules/postgres.py

```diff
@@ -182,9 +223,12 @@ def _run_initdb(name,
 
     if user is None:
         user = runas
-
+    if salt.utils.which('initdb') is None:
+        _INITDB_BIN = _check_initdb_bin()
+    else:
+        _INITDDB_BIN = salt.utils.which('initdb')
```

Comment:
It appears that `postgres.initdb_bin` is only used if `initdb` could not be
found in the path. Isn't the opposite intended?

I also heartily recommend working the PG installation path so that only one
configuration option needs to be specified. Something like this:

```python
def _find_pg_binary(util):
    pg_bin_dir = __salt__['config.option']('postgres.bin_dir')
    if pg_bin_dir:
        return os.path.join(pg_bin_dir, util)
    else:
        return salt.utils.which(util)
```